nf_core/pipelines/lint/nf_test_content.py

```diff
@@ -0,0 +1,191 @@
+            lint:
+                nf_test_content:
+                    - outdir
+                    - versions.yml
+                    - modules_testdata_base_path
+                    - pipelines_testdata_base_path
+                    - resourceLimits
+                    - testsDir
+                    - workDir
+                    - configFile
+    """
+    passed = []
+    failed = []
+    ignored = []
+
+    _, pipeline_conf = load_tools_config(self.wf_path)
+    lint_conf = getattr(pipeline_conf, "lint", None) or None
+    nf_test_content_conf = getattr(lint_conf, "nf_test_content", None) or []
+
+    # Content of *.nf.test files
+    test_fns = list(Path(self.wf_path, "tests").glob("*.nf.test"))
+    outdir_pass = False
+    versions_pass = False
+    for test_fn in test_fns:
+        with open(test_fn) as fh:
+            for line in fh:
+                # Check outdir is defined
+                if "when" in line and "outdir" not in nf_test_content_conf:
+                    while "}\n" not in line:
+                        line = next(fh)
+                        if re.search(r"outdir *= *[\"']\${?outputDir}?[\"']", line):
+                            passed.append(f"'{test_fn}' contains `outdir` parameter")
+                            outdir_pass = True
+                            break
+                # Check versions.yml is included in the snapshot
+                if "versions.yml" in line and "versions.yml" not in nf_test_content_conf:
+                    passed.append(f"'{test_fn}' snapshots a 'versions.yml' file")
+                    versions_pass = True
+        if not outdir_pass and "outdir" not in nf_test_content_conf:
+            failed.append(
+                f"""'{test_fn}' does not contain `outdir` parameter, it should contain `outdir = "$outputDir"`"""
+            )
+        elif "outdir" in nf_test_content_conf:
+            ignored.append(f"'{test_fn}' checking `outdir` parameter ignored")
+        if not versions_pass and "versions.yml" not in nf_test_content_conf:
+            failed.append(f"'{test_fn}' snaphsots a 'versions.yml' file")
+        elif "versions.yml" in nf_test_content_conf:
+            ignored.append(f"'{test_fn}' checking `versions.yml` in snapshot ignored")
+
+    # Content of nextflow.config file
+    conf_fn = Path(self.wf_path, "tests", "nextflow.config")
+    modules_testdata_base_path_pass = False
+    pipelines_testdata_base_path_pass = False
+    cpus_pass = False
+    memory_pass = False
+    time_pass = False
+
+    with open(conf_fn) as fh:
+        for line in fh:
+            line = line.strip()
+            if "modules_testdata_base_path" in line and "modules_testdata_base_path" not in nf_test_content_conf:
+                passed.append(f"'{conf_fn}' contains `modules_testdata_base_path`")
+                modules_testdata_base_path_pass = True
+            if "pipelines_testdata_base_path" in line and "pipelines_testdata_base_path" not in nf_test_content_conf:
```

Comment:
we don't have `pipelines_testdata_base_path` in the template, is it mandatory? maybe we don't need to lint for it

Reply:
added to the template